app/src/main/java/vn/sunasterisk/music_67/ui/detailplaying/DetailPlayingActivity.kt

```diff
@@ -1,39 +1,207 @@
+			override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+			}
+
+			override fun onStartTrackingTouch(seekBar: SeekBar?) {
+				playingTracksService.pause()
+				playToPause()
+			}
+
+			override fun onStopTrackingTouch(seekBar: SeekBar) {
+				playingTracksService.seek(seekBar.progress)
+				pauseToPlay()
+			}
+		})
+	}
+
+	override fun onDestroy() {
+		super.onDestroy()
+		unbindService(serviceConnection)
+	}
+
+	override fun onClick(v: View?) {
+		when (v) {
+			imageBefore -> onBackPressed()
+		}
+	}
+
+	override fun completeListener() {
+		nextTrack()
+	}
+
+	private fun updateState() {
+		handler = Handler(Looper.getMainLooper()) {
+			when (it.what) {
+				MSG_HANDLER -> {
+					seekBarPlaying.progress = it.arg1
+					textCurrentTime.text = TimeConvert.convertMilisecondToFormatTime(it.arg1.toLong())
+					if (it.arg2 == 1) {
+						pauseToPlay()
+					} else {
+						playToPause()
+					}
+					return@Handler true
+				}
+				else -> return@Handler false
+			}
+		}
+	}
+
+	private fun getState() {
+		thread = Thread(Runnable {
+			val total = playingTracksService.getDuration().toInt()
+			seekBarPlaying.max = total
+			var cur = playingTracksService.getCurrentPoint()
+			while (cur < total) {
+				val message = Message()
+				message.apply {
+					what = MSG_HANDLER
+					arg1 = cur
+					when (playingTracksService.isPlaying()) {
+						true -> arg2 = 1
+						false -> arg2 = 0
+					}
+				}
+				handler.sendMessage(message)
+				try {
+					Thread.sleep(500)
+					cur = playingTracksService.getCurrentPoint()
+				} catch (e: InterruptedException) {
+					e.printStackTrace()
+				}
+			}
+		})
+		thread.start()
+	}
+
+	private fun updateUI(track: Track) {
+		getUI(track)
+		pauseToPlay()
+		getState()
+		updateState()
 	}
 
 	private fun getContentIntent() {
-		val trackTitle = intent.getStringExtra(TrackAttributes.TITLE)
-		val trackArtist = intent.getStringExtra(TrackAttributes.ARTIST)
-		val durationTrack = intent.getLongExtra(TrackAttributes.DURATION, DEFAULT_VALUE.toLong())
-		val imageTrack = intent.getStringExtra(TrackAttributes.ARTWORK_URL)
-		textTitle.text = trackTitle
-		textArtist.text = trackArtist
-		textDuration.text = TimeConvert.convertMilisecondToFormatTime(durationTrack)
+		val trackReceived = intent.getParcelableExtra(TrackAttributes.TRACK) as Track
+		getUI(trackReceived)
+	}
+
+	private fun getUI(track: Track) {
+		textTitle.text = track.title
+		textArtist.text = track.artist
+		textDuration.text = TimeConvert.convertMilisecondToFormatTime(track.duration!!)
 		textCurrentTime.text = TimeConvert.convertMilisecondToFormatTime(0)
-		Glide.with(this)
-				.load(imageTrack)
+		Glide.with(applicationContext)
+				.load(track.artworkUrl)
 				.error(R.drawable.icon_app)
 				.into(imageBackground)
-		Glide.with(this)
-				.load(imageTrack)
+		Glide.with(applicationContext)
+				.load(track.artworkUrl)
 				.error(R.drawable.icon_app)
```

Comment:
Nên set cả .fallback(R.drawable.ic_no_image). Use fallback image resource when the url can be null